### DISCUSSION TITLE
`.config` dir support tracker

### DISCUSSION THREAD

**Discussion by pi0** [pi0/config-dir#6] [General]
> [!NOTE]
> Here is a list of tools that either support `.config` or plan to do.
> 
> See [previous efforts](https://github.com/pi0/config-dir/discussions/5) discussion for other relevant efforts. 

## ✅ Support `.config/`

- [c12](https://github.com/unjs/c12)
- [cosmiconfig](https://github.com/cosmiconfig/cosmiconfig)
- [nuxt](https://nuxt.com)
- [nitro](https://nitro.build)
- [unbuild](https://unbuild.unjs.io)
- [marklint](https://github.com/Marcisbee/marklint) 
- [rubocop](https://github.com/rubocop/rubocop)
- [mise](https://github.com/jdx/mise)
- [goreleaser](https://github.com/goreleaser/goreleaser)
- [vue-mess-detector](https://github.com/rrd108/vue-mess-detector)
- [cargo-nextest](https://nexte.st)
- [lefthook](https://evilmartians.github.io/lefthook/)
- [hk](https://github.com/jdx/hk)
- [rumdl](https://github.com/rvben/rumdl)
- [cspell](https://cspell.org/docs/getting-started#1-create-a-configuration-file)
- [air-verse](https://github.com/air-verse/air)
- [ryl](https://github.com/owenlamont/ryl)

## 💬 Discussions

Open:

- vite: https://github.com/vitejs/vite/discussions/16527
- vitest: https://github.com/vitest-dev/vitest/issues/5204
- proload: https://github.com/natemoo-re/proload/issues/34
- unifiedjs: https://github.com/unifiedjs/unified-engine/issues/72
- biome: https://github.com/biomejs/biome/discussions/1833
- golangci-lint: https://github.com/golangci/golangci-lint/issues/4554
- dprint: https://github.com/dprint/dprint/issues/938
- lintrunner: https://github.com/suo/lintrunner/pull/69
- crate-ci: https://github.com/crate-ci/committed/issues/419
- prisma: https://github.com/prisma/prisma/issues/26369
- github: https://github.com/orgs/community/discussions/178064
- editorconfig: https://github.com/editorconfig/editorconfig/issues/568

Closed: (**mostly suggesting users to override config path**)

- eslint: https://github.com/eslint/eslint/issues/18294
- drizzle: https://github.com/drizzle-team/drizzle-orm/issues/1896
- mkdocs: https://github.com/mkdocs/mkdocs/issues/3601
- docker compose: https://github.com/docker/compose/issues/11640
- ruff: https://github.com/astral-sh/ruff/issues/10493

---

_(please comment to add more links here)_

**Comment by remcohaszing** [comment]
There's a proposal for adding support to `unified-engine` (remark, rehype, retext).

https://github.com/unifiedjs/unified-engine/issues/72

**Comment by kvnwolf** [comment]
I opened a proposal for [Biome](https://biomejs.dev): https://github.com/biomejs/biome/discussions/1833

**Comment by Yizack** [comment]
I just opened a proposal for [Drizzle](https://orm.drizzle.team/) at https://github.com/drizzle-team/drizzle-orm/issues/1896

**Comment by bersace** [comment]
https://github.com/goreleaser/goreleaser/issues/4716

**Comment by bersace** [comment]
https://github.com/golangci/golangci-lint/issues/4554

**Comment by bersace** [comment]
https://github.com/mkdocs/mkdocs/issues/3601

**Comment by bersace** [comment]
https://github.com/moby/moby/issues/47600

**Comment by bersace** [comment]
https://github.com/astral-sh/ruff/issues/10493

**Comment by Splines** [comment]
I opened a change request for ESLint:

- https://github.com/eslint/eslint/issues/18294.

**Edit**: turns out we can just use the `--config` cli flag in ESLint to specify a path to the config file. See [this comment](https://github.com/eslint/eslint/issues/18294#issuecomment-2044091313).

In the case you're using VSCode, you might want to put this in your `.vscode/settings.json` for your workspace:

```json
"eslint.options": {
    "overrideConfigFile": ".config/eslint.mjs"
}
```

See the docs for the [`overrideConfigFile` param](https://eslint.org/docs/latest/integrate/nodejs-api#parameters).

**Reply by TKDev7** [reply]
I wish if eslint can find it by default.

**Comment by Splines** [comment]
`rubocop`, the Ruby linter, has implemented it 😍, see this PR:
- https://github.com/rubocop/rubocop/pull/12700

And the [docs](https://docs.rubocop.org/rubocop/configuration.html#config-file-locations).

**Comment by bersace** [comment]
https://github.com/jdx/mise supports .config !

**Comment by rrd108** [comment]
Vue Mess Detector supports `.config/` since 0.61.0

Thanks for the proposal

**Comment by scop** [comment]
RFE for [dprint](https://dprint.dev): https://github.com/dprint/dprint/issues/938

**Comment by scop** [comment]
[cargo-nextest](https://nexte.st) : [supports `.config`](https://nexte.st/docs/configuration/) :+1:

**Comment by scop** [comment]
RFE for [committed](https://github.com/crate-ci/committed): https://github.com/crate-ci/committed/issues/419

**Comment by scop** [comment]
RFE/WIP pull request for [lefthook](https://evilmartians.github.io/lefthook/): https://github.com/evilmartians/lefthook/pull/933

**Reply by scop** [reply]
This got implemented in https://github.com/evilmartians/lefthook/pull/1017, said to appear in lefthook 1.11.12 :+1:

**Reply by scop** [reply]
lefthook now supports .config in >= 1.11.12

**Comment by e965** [comment]
Proposal for Prisma - https://github.com/prisma/prisma/issues/26369

**Reply by infogulch** [reply]
Looks like prisma was implemented. https://github.com/prisma/prisma/pull/27672

**Comment by flx-sta** [comment]
@pi0 lefthook support is marked as "ongoing" but has been added with the `1.11.12` release.

It's mentioned in their documentation:
https://lefthook.dev/configuration/index.html

**Reply by scop** [reply]
Yeah, that was noted in discussions above, https://github.com/pi0/config-dir/discussions/6#discussioncomment-12972892

Anyway would be good to bring the summary up to date.

**Reply by flx-sta** [reply]
Whops. Thanks for bringing that to my attentiont

**Reply by pi0** [reply]
thnx updated!

**Comment by jerriclynsjohn** [comment]
Was really hoping to see this implemented with Prettier

**Reply by infogulch** [reply]
Did you submit a request?

**Comment by infogulch** [comment]
Submitted request for `.github` directory:

[Look for "dot github" / .github configuration files in .config/github #178064](https://github.com/orgs/community/discussions/178064)

**Comment by scop** [comment]
RFE for [hk](https://hk.jdx.dev): https://github.com/jdx/hk/discussions/405

**Reply by scop** [reply]
Implemented in hk >= 1.21.0, https://github.com/jdx/hk/pull/407

**Comment by scop** [comment]
RFE for [rumdl](https://github.com/rvben/rumdl): https://github.com/rvben/rumdl/issues/162

**Reply by scop** [reply]
Implemented in rumdl >= 0.0.182, https://github.com/rvben/rumdl/issues/162#issuecomment-3576593474

**Comment by infogulch** [comment]
Request for editorconfig: https://github.com/editorconfig/editorconfig/issues/568

**Comment by carlosepcc** [comment]
Prisma
I found out about this proposal thanks to Prisma, linking it here:
https://www.prisma.io/docs/orm/reference/prisma-config-reference#engine

**Comment by scop** [comment]
RFE for [lychee](https://lychee.cli.rs/): https://github.com/lycheeverse/lychee/issues/1951

**Comment by Splines** [comment]
Proposal for herb: https://github.com/marcoroth/herb/issues/1176

**Comment by o-l-a-v** [comment]
Proposal for the PowerShell linter "PSScriptAnalyzer":

* <https://github.com/PowerShell/PSScriptAnalyzer/issues/2157>

**Comment by teamwolfyta** [comment]
Proposal for Treefmt: 
- Issue: https://github.com/numtide/treefmt/issues/668
- PR: https://github.com/numtide/treefmt/pull/669

**Comment by natescherer** [comment]
Proposal for Tombi: https://github.com/tombi-toml/tombi/issues/1674

**Reply by natescherer** [reply]
A very impressive 7 hour turnaround on this, just confirmed it works! In Tombi >= 0.9.14

**Comment by natescherer** [comment]
Proposal for All Contributors: https://github.com/all-contributors/cli/issues/499

**Comment by natescherer** [comment]
Proposal for Renovate: https://github.com/renovatebot/renovate/discussions/42367

(Upvotes on that would probably help, there are a LOT of feature requests for Renovate!)

**Comment by bersace** [comment]
Proposal for Zensical: https://github.com/zensical/backlog/issues/128

**Comment by bersace** [comment]
Proposal for Zed editor: https://github.com/zed-industries/zed/discussions/53289

**Comment by lachieh** [comment]
Proposal for portless: https://github.com/vercel-labs/portless/issues/268

**Comment by o-l-a-v** [comment]
Proposal for ryl: <https://github.com/owenlamont/ryl/issues/218>

**Comment by bersace** [comment]
Next release of Go air will support .config. https://github.com/air-verse/air/pull/716#event-25808959496

**Comment by lonix1** [comment]
Proposal for cspell: https://github.com/streetsidesoftware/cspell/issues/8921

**Reply by lonix1** [reply]
@pi0 CSpell actually already supports `.config/`... Please add to list?

**Comment by xzfc** [comment]
RFE for ast-grep: https://github.com/ast-grep/ast-grep/issues/2840

**Comment by niklaskorz** [comment]
Feature request for Gram: https://codeberg.org/GramEditor/gram/wiki/Support-.config%2Fgram-for-workspace-configuration

**Comment by mahlzahn** [comment]
Typos: https://github.com/crate-ci/typos/issues/1445
